app/models/base_channel.js

```diff
@@ -80,7 +81,9 @@ export default Ember.Object.extend({
   },
 
   addUser(username) {
-    this.get('userList').pushObject(username);
+    if (isEmpty(this.get('userList').find(name => name === username))) {
```

Comment:
Even better. Then we could do [`if (this.get('userList').includes(username))`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes?v=control).